### ISSUE TITLE
Updated PKC 2.7.10 crashes/freezes Kodi on exit on my FireTV 2 (But not Fire Stick 4k)

### ISSUE THREAD

**Issue by jhr1948** [croneter/PlexKodiConnect#812]
I have Kodi 18 (also tried 18.1 with same result) on a FireTV 2 and the newest PKC 2.7.10 hangs indefinitely upon exit.  I have to go directly to FireTV menu and go in apps to force stop.

I have the newest Firestick 4k and PCK 2.7.10 is fine.

Uploaded Log.

[kodi.log](https://github.com/croneter/PlexKodiConnect/files/3025816/kodi.log)
## Help yourself
* I did try to restart Kodi :-)
* I checked the [PKC Frequently Asked Questions on the PKC wiki](https://github.com/croneter/PlexKodiConnect/wiki/faq)
* I did try to reset the Kodi database by going to `PKC Settings -> Advanced -> "Reset the database and optionally reset PlexKodiConnect"` and then hitting YES, NO
* I did check the [existing issues on Github](https://github.com/croneter/PlexKodiConnect/issues)

## You need to attach a KODI LOG FILE!
A Kodi debug log file is needed that you recorded while you reproduced the bug. **Do clean your log of all Plex tokens (="Plex passwords")!!!** .
1. Activate Kodi's debug logging by going to the Kodi `Settings` -> `System` -> `Logging`. Then toggle the `Enable debug logging` setting.
2. Restart Kodi to start with a "fresh" log file.
3. Reproduce the bug.
4. Follow the [Kodi instructions](http://kodi.wiki/view/Log_file/Easy) to grab/share the Kodi log file. Usually only `kodi.log` is needed
    * You can [find the log file here](http://kodi.wiki/view/Log_file/Advanced#Location)
5. **Delete all references to any of your Plex tokens** by searching for `X-Plex-Token` and `accesstoken` and replacing the strings just after that!
    * It's easiest if you copy your token, then use Search&Replace for the entire log file
    * You don't want others to have access to your Plex installation....
6. Drop your log file here in this issue. Or use a free pasting-service like https://pastebin.com and include the link to it here
    
I am aware that I can delete Plex tokens that I accidentially posted by following the [instructions on the PKC wiki](https://github.com/croneter/PlexKodiConnect/wiki/How-to-Report-A-Bug#i-published-my-plex-token-to-some-forum-or-github-anyone-can-now-access-my-plex-server)

**Comment by croneter** [comment]
Sorry, but this is a Kodi issue. Everything seems perfectly fine looking at the log. Kodi **IS** a PITA when shutting down, down. 

Please take this up with the Kodi team and share the discussion/bug/... here...

**Comment by croneter** [comment]
PKC **beta** version 2.7.12, available in a couple of minutes, *might* help. Thanks for your feedback

**Comment by jhr1948** [comment]
Thank you for the help.  I understand it could be linked to kodi, but the only thing changed was PKC.  I've actually experimented over and a over (fresh installs, no skin change, kept simple) and I get the same hanging exit only with PKC 2.7.10 on the Fire TV 2.  

The Fire stick 4k I have doesn't have this problem with 2.7.10.  I rolled back PKC 2.7.4 on the FireTV 2 and it works properly.  Did you not see anything fishy in the log?
